spring-boot-project/spring-boot/src/main/java/org/springframework/boot/ansi/AnsiColors.java

```diff
@@ -32,30 +31,31 @@
  * @author Ruben Dijkstra
  * @author Phillip Webb
  * @author Michael Simons
+ * @author TomXin
  * @since 1.4.0
  */
 public final class AnsiColors {
 
-	private static final Map<AnsiElement, LabColor> ANSI_COLOR_MAP;
+	private static final Map<AnsiColorWrapper, LabColor> ANSI_COLOR_MAP;
 
 	static {
-		Map<AnsiColor, LabColor> colorMap = new EnumMap<>(AnsiColor.class);
-		colorMap.put(AnsiColor.BLACK, new LabColor(0x000000));
-		colorMap.put(AnsiColor.RED, new LabColor(0xAA0000));
-		colorMap.put(AnsiColor.GREEN, new LabColor(0x00AA00));
-		colorMap.put(AnsiColor.YELLOW, new LabColor(0xAA5500));
-		colorMap.put(AnsiColor.BLUE, new LabColor(0x0000AA));
-		colorMap.put(AnsiColor.MAGENTA, new LabColor(0xAA00AA));
-		colorMap.put(AnsiColor.CYAN, new LabColor(0x00AAAA));
-		colorMap.put(AnsiColor.WHITE, new LabColor(0xAAAAAA));
-		colorMap.put(AnsiColor.BRIGHT_BLACK, new LabColor(0x555555));
-		colorMap.put(AnsiColor.BRIGHT_RED, new LabColor(0xFF5555));
-		colorMap.put(AnsiColor.BRIGHT_GREEN, new LabColor(0x55FF00));
-		colorMap.put(AnsiColor.BRIGHT_YELLOW, new LabColor(0xFFFF55));
-		colorMap.put(AnsiColor.BRIGHT_BLUE, new LabColor(0x5555FF));
-		colorMap.put(AnsiColor.BRIGHT_MAGENTA, new LabColor(0xFF55FF));
-		colorMap.put(AnsiColor.BRIGHT_CYAN, new LabColor(0x55FFFF));
-		colorMap.put(AnsiColor.BRIGHT_WHITE, new LabColor(0xFFFFFF));
+		Map<AnsiColorWrapper, LabColor> colorMap = new LinkedHashMap<>(16);
+		colorMap.put(new AnsiColorWrapper(30, BitDepth.FOUR), new LabColor(0x000000));
+		colorMap.put(new AnsiColorWrapper(31, BitDepth.FOUR), new LabColor(0xAA0000));
```

Comment:
It is better to leave constants instead of numbers

Reply:
> It is better to leave constants instead of numbers
Yes, this is a good suggestion, I will improve this tomorrow

Reply:
> It is better to leave constants instead of numbers

I solved it